### ISSUE TITLE
Flyway migration errors cause quarkus unit tests to be ignored and succeed the tests and class-loader issue hides the migration script error

### ISSUE THREAD

**Issue by FroMage** [quarkusio/quarkus#43926]
### Describe the bug

Omitting a semi-colon in my flyway migration script in a quarkus unit test causes an exception being thrown while handling the syntax exception, which seems to be due to a class loader not finding the migration script, which hides the original syntax error.

Additionally, this causes all tests to be ignored, and the build to succeed.

```java
package io.quarkus.hibernate.orm.panache.deployment.test.multitenant;

import jakarta.persistence.Entity;

import org.assertj.core.api.Assertions;
import org.jboss.shrinkwrap.api.asset.StringAsset;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.RegisterExtension;

import io.quarkus.hibernate.orm.panache.PanacheEntity;
import io.quarkus.test.QuarkusUnitTest;

public class FlywayIssue1 {
    @RegisterExtension
    static QuarkusUnitTest runner = new QuarkusUnitTest()
            .withApplicationRoot((jar) -> jar
                    .addClasses(MyEntity.class)
                    .add(new StringAsset("""
                            create sequence MyEntity_SEQ start with 1 increment by 50 /*semi-colon omitted on purpose*/
                                  create table MyEntity (
                                      id bigint not null,
                                      primary key (id)
                                  );
                            """), "database/multitenant/V1.0.0__create.sql")
                    .add(new StringAsset("""
                                  # This only works for the read-only DB
                                  quarkus.hibernate-orm.database.generation=none
                                  quarkus.hibernate-orm.datasource=read-only

                                  quarkus.datasource.read-only.db-kind=h2
                                  quarkus.datasource.read-only.jdbc.url=jdbc:h2:mem:first;DB_CLOSE_DELAY=-1

                                  quarkus.flyway.read-only.locations=database/multitenant
                                  quarkus.flyway.read-only.migrate-at-start=true
                            """), "application.properties"));

    @Entity
    public static class MyEntity extends PanacheEntity {
    }

    @Test
    public void panacheOperations() {
        Assertions.fail();
    }

}
```

Build output:

```
[INFO] Running io.quarkus.hibernate.orm.panache.deployment.test.multitenant.FlywayIssue1
2024-10-17 11:48:35,309 WARN  [io.qua.dat.dep.dev.DevServicesDatasourceProcessor] (build-9) Unable to determine a database type for default datasource
2024-10-17 11:48:37,171 INFO  [org.fly.cor.FlywayExecutor] (main) Database: jdbc:h2:mem:first (H2 2.3)
2024-10-17 11:48:37,176 WARN  [org.fly.cor.int.dat.bas.Database] (main) Flyway upgrade recommended: H2 2.3.230 is newer than this version of Flyway and support has not been tested. The latest supported version of H2 is 2.2.224.
2024-10-17 11:48:37,189 INFO  [org.fly.cor.int.sch.JdbcTableSchemaHistory] (main) Schema history table "PUBLIC"."flyway_schema_history" does not exist yet
2024-10-17 11:48:37,191 INFO  [org.fly.cor.int.com.DbValidate] (main) Successfully validated 1 migration (execution time 00:00.007s)
2024-10-17 11:48:37,196 INFO  [org.fly.cor.int.sch.JdbcTableSchemaHistory] (main) Creating Schema History table "PUBLIC"."flyway_schema_history" ...
2024-10-17 11:48:37,221 INFO  [org.fly.cor.int.com.DbMigrate] (main) Current version of schema "PUBLIC": << Empty Schema >>
2024-10-17 11:48:37,225 INFO  [org.fly.cor.int.com.DbMigrate] (main) Migrating schema "PUBLIC" to version "1.0.0 - create"
2024-10-17 11:48:37,233 ERROR [org.fly.cor.int.com.DbMigrate] (main) Migration of schema "PUBLIC" to version "1.0.0 - create" failed! Please restore backups and roll back database and code!
2024-10-17 11:48:37,296 WARNING [org.jun.pla.lau.cor.CompositeTestExecutionListener] (main) TestExecutionListener [org.apache.maven.surefire.junitplatform.RunListenerAdapter] threw exception for method: executionFinished(TestIdentifier [uniqueId = [engine:junit-jupiter]/[class:io.quarkus.hibernate.orm.panache.deployment.test.multitenant.FlywayIssue1], parentId = [engine:junit-jupiter], displayName = 'FlywayIssue1', legacyReportingName = 'io.quarkus.hibernate.orm.panache.deployment.test.multitenant.FlywayIssue1', source = ClassSource [className = 'io.quarkus.hibernate.orm.panache.deployment.test.multitenant.FlywayIssue1', filePosition = null], tags = [], type = CONTAINER], TestExecutionResult [status = FAILED, throwable = java.lang.RuntimeException: java.lang.RuntimeException: Failed to start quarkus]): org.flywaydb.core.api.FlywayException: Unable to find resource on disk: database/multitenant/V1.0.0__create.sql
	at org.flywaydb.core.internal.resource.classpath.ClassPathResource.getAbsolutePathOnDisk(ClassPathResource.java:85)
	at org.flywaydb.core.internal.sqlscript.FlywaySqlScriptException.getMessage(FlywaySqlScriptException.java:79)
	at java.base/java.lang.Throwable.getLocalizedMessage(Throwable.java:397)
	at java.base/java.lang.Throwable.toString(Throwable.java:496)
	at java.base/java.lang.String.valueOf(String.java:4220)
	at java.base/java.lang.StringBuilder.append(StringBuilder.java:173)
	at java.base/java.lang.Throwable.printEnclosedStackTrace(Throwable.java:710)
	at java.base/java.lang.Throwable.printEnclosedStackTrace(Throwable.java:724)
	at java.base/java.lang.Throwable.printEnclosedStackTrace(Throwable.java:724)
	at java.base/java.lang.Throwable.printStackTrace(Throwable.java:682)
	at java.base/java.lang.Throwable.printStackTrace(Throwable.java:736)
	at org.apache.maven.surefire.report.PojoStackTraceWriter.writeTraceToString(PojoStackTraceWriter.java:52)
	at org.apache.maven.surefire.booter.spi.EventChannelEncoder.toStackTrace(EventChannelEncoder.java:395)
	at org.apache.maven.surefire.booter.spi.EventChannelEncoder.access$000(EventChannelEncoder.java:76)
	at org.apache.maven.surefire.booter.spi.EventChannelEncoder$StackTrace.<init>(EventChannelEncoder.java:407)
	at org.apache.maven.surefire.booter.spi.EventChannelEncoder.encode(EventChannelEncoder.java:336)
	at org.apache.maven.surefire.booter.spi.EventChannelEncoder.encode(EventChannelEncoder.java:273)
	at org.apache.maven.surefire.booter.spi.EventChannelEncoder.testError(EventChannelEncoder.java:152)
	at org.apache.maven.surefire.api.booter.ForkingRunListener.testError(ForkingRunListener.java:79)
	at org.apache.maven.surefire.junitplatform.RunListenerAdapter.executionFinished(RunListenerAdapter.java:145)
	at org.junit.platform.launcher.core.CompositeTestExecutionListener.lambda$executionFinished$10(CompositeTestExecutionListener.java:73)
	at org.junit.platform.launcher.core.CompositeTestExecutionListener.lambda$notifyEach$19(CompositeTestExecutionListener.java:102)
	at org.junit.platform.commons.util.CollectionUtils.forEachInReverseOrder(CollectionUtils.java:221)
	at org.junit.platform.launcher.core.IterationOrder$2.forEach(IterationOrder.java:30)
	at org.junit.platform.launcher.core.CompositeTestExecutionListener.notifyEach(CompositeTestExecutionListener.java:100)
	at org.junit.platform.launcher.core.CompositeTestExecutionListener.executionFinished(CompositeTestExecutionListener.java:72)
	at org.junit.platform.launcher.core.ExecutionListenerAdapter.executionFinished(ExecutionListenerAdapter.java:56)
	at org.junit.platform.launcher.core.CompositeEngineExecutionListener.lambda$executionFinished$6(CompositeEngineExecutionListener.java:59)
	at org.junit.platform.launcher.core.CompositeEngineExecutionListener.lambda$notifyEach$11(CompositeEngineExecutionListener.java:74)
	at org.junit.platform.commons.util.CollectionUtils.forEachInReverseOrder(CollectionUtils.java:221)
	at org.junit.platform.launcher.core.IterationOrder$2.forEach(IterationOrder.java:30)
	at org.junit.platform.launcher.core.CompositeEngineExecutionListener.notifyEach(CompositeEngineExecutionListener.java:72)
	at org.junit.platform.launcher.core.CompositeEngineExecutionListener.executionFinished(CompositeEngineExecutionListener.java:58)
	at org.junit.platform.launcher.core.DelegatingEngineExecutionListener.executionFinished(DelegatingEngineExecutionListener.java:46)
	at org.junit.platform.launcher.core.StackTracePruningEngineExecutionListener.executionFinished(StackTracePruningEngineExecutionListener.java:46)
	at org.junit.platform.launcher.core.DelegatingEngineExecutionListener.executionFinished(DelegatingEngineExecutionListener.java:46)
	at org.junit.platform.launcher.core.OutcomeDelayingEngineExecutionListener.executionFinished(OutcomeDelayingEngineExecutionListener.java:63)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.reportCompletion(NodeTestTask.java:195)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:100)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:198)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:169)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:93)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:58)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:141)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:57)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:103)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:85)
	at org.junit.platform.launcher.core.DelegatingLauncher.execute(DelegatingLauncher.java:47)
	at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:56)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:184)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:148)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:122)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)

2024-10-17 11:48:37,302 WARNING [org.jun.pla.lau.cor.CompositeTestExecutionListener] (main) TestExecutionListener [com.gradle.maven.scan.extension.test.listener.junitplatform.NestedExecutionSafeDelegatingTestExecutionListener] threw exception for method: executionFinished(TestIdentifier [uniqueId = [engine:junit-jupiter]/[class:io.quarkus.hibernate.orm.panache.deployment.test.multitenant.FlywayIssue1], parentId = [engine:junit-jupiter], displayName = 'FlywayIssue1', legacyReportingName = 'io.quarkus.hibernate.orm.panache.deployment.test.multitenant.FlywayIssue1', source = ClassSource [className = 'io.quarkus.hibernate.orm.panache.deployment.test.multitenant.FlywayIssue1', filePosition = null], tags = [], type = CONTAINER], TestExecutionResult [status = FAILED, throwable = java.lang.RuntimeException: java.lang.RuntimeException: Failed to start quarkus]): org.flywaydb.core.api.FlywayException: Unable to find resource on disk: database/multitenant/V1.0.0__create.sql
	at org.flywaydb.core.internal.resource.classpath.ClassPathResource.getAbsolutePathOnDisk(ClassPathResource.java:85)
	at org.flywaydb.core.internal.sqlscript.FlywaySqlScriptException.getMessage(FlywaySqlScriptException.java:79)
	at com.gradle.maven.scan.extension.test.event.internal.ThrowableConverter.convert(SourceFile:20)
	at com.gradle.maven.scan.extension.test.event.internal.ThrowableConverter.convert(SourceFile:23)
	at com.gradle.maven.scan.extension.test.event.internal.ThrowableConverter.convert(SourceFile:23)
	at com.gradle.maven.scan.extension.test.event.internal.ThrowableConverter.convert(SourceFile:23)
	at com.gradle.maven.scan.extension.test.event.TestFinishedEvent.with(SourceFile:36)
	at com.gradle.maven.scan.extension.test.listener.obfuscated.i.a.a(SourceFile:54)
	at com.gradle.maven.scan.extension.test.listener.junitplatform.a.a(SourceFile:148)
	at com.gradle.maven.scan.extension.test.listener.junitplatform.a.executionFinished(SourceFile:129)
	at com.gradle.maven.scan.extension.test.listener.junitplatform.b.executionFinished(SourceFile:53)
	at com.gradle.maven.scan.extension.test.listener.junitplatform.NestedExecutionSafeDelegatingTestExecutionListener.executionFinished(SourceFile:57)
	at org.junit.platform.launcher.core.CompositeTestExecutionListener.lambda$executionFinished$10(CompositeTestExecutionListener.java:73)
	at org.junit.platform.launcher.core.CompositeTestExecutionListener.lambda$notifyEach$19(CompositeTestExecutionListener.java:102)
	at org.junit.platform.commons.util.CollectionUtils.forEachInReverseOrder(CollectionUtils.java:221)
	at org.junit.platform.launcher.core.IterationOrder$2.forEach(IterationOrder.java:30)
	at org.junit.platform.launcher.core.CompositeTestExecutionListener.notifyEach(CompositeTestExecutionListener.java:100)
	at org.junit.platform.launcher.core.CompositeTestExecutionListener.executionFinished(CompositeTestExecutionListener.java:72)
	at org.junit.platform.launcher.core.ExecutionListenerAdapter.executionFinished(ExecutionListenerAdapter.java:56)
	at org.junit.platform.launcher.core.CompositeEngineExecutionListener.lambda$executionFinished$6(CompositeEngineExecutionListener.java:59)
	at org.junit.platform.launcher.core.CompositeEngineExecutionListener.lambda$notifyEach$11(CompositeEngineExecutionListener.java:74)
	at org.junit.platform.commons.util.CollectionUtils.forEachInReverseOrder(CollectionUtils.java:221)
	at org.junit.platform.launcher.core.IterationOrder$2.forEach(IterationOrder.java:30)
	at org.junit.platform.launcher.core.CompositeEngineExecutionListener.notifyEach(CompositeEngineExecutionListener.java:72)
	at org.junit.platform.launcher.core.CompositeEngineExecutionListener.executionFinished(CompositeEngineExecutionListener.java:58)
	at org.junit.platform.launcher.core.DelegatingEngineExecutionListener.executionFinished(DelegatingEngineExecutionListener.java:46)
	at org.junit.platform.launcher.core.StackTracePruningEngineExecutionListener.executionFinished(StackTracePruningEngineExecutionListener.java:46)
	at org.junit.platform.launcher.core.DelegatingEngineExecutionListener.executionFinished(DelegatingEngineExecutionListener.java:46)
	at org.junit.platform.launcher.core.OutcomeDelayingEngineExecutionListener.executionFinished(OutcomeDelayingEngineExecutionListener.java:63)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.reportCompletion(NodeTestTask.java:195)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:100)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:198)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:169)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:93)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:58)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:141)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:57)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:103)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:85)
	at org.junit.platform.launcher.core.DelegatingLauncher.execute(DelegatingLauncher.java:47)
	at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:56)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:184)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:148)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:122)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)

[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- jar:3.3.0:jar (default-jar) @ quarkus-hibernate-orm-panache-deployment ---
[INFO] Building jar: /home/stephane/src/java-eclipse/quarkus/extensions/panache/hibernate-orm-panache/deployment/target/quarkus-hibernate-orm-panache-deployment-999-SNAPSHOT.jar
[INFO] 
[INFO] --- source:3.2.1:jar-no-fork (attach-sources) @ quarkus-hibernate-orm-panache-deployment ---
[INFO] Building jar: /home/stephane/src/java-eclipse/quarkus/extensions/panache/hibernate-orm-panache/deployment/target/quarkus-hibernate-orm-panache-deployment-999-SNAPSHOT-sources.jar
[INFO] 
[INFO] --- install:3.1.1:install (default-install) @ quarkus-hibernate-orm-panache-deployment ---
[INFO] Installing /home/stephane/src/java-eclipse/quarkus/extensions/panache/hibernate-orm-panache/deployment/pom.xml to /home/stephane/.m2/repository/io/quarkus/quarkus-hibernate-orm-panache-deployment/999-SNAPSHOT/quarkus-hibernate-orm-panache-deployment-999-SNAPSHOT.pom
[INFO] Installing /home/stephane/src/java-eclipse/quarkus/extensions/panache/hibernate-orm-panache/deployment/target/quarkus-hibernate-orm-panache-deployment-999-SNAPSHOT.jar to /home/stephane/.m2/repository/io/quarkus/quarkus-hibernate-orm-panache-deployment/999-SNAPSHOT/quarkus-hibernate-orm-panache-deployment-999-SNAPSHOT.jar
[INFO] Installing /home/stephane/src/java-eclipse/quarkus/extensions/panache/hibernate-orm-panache/deployment/target/quarkus-hibernate-orm-panache-deployment-999-SNAPSHOT-sources.jar to /home/stephane/.m2/repository/io/quarkus/quarkus-hibernate-orm-panache-deployment/999-SNAPSHOT/quarkus-hibernate-orm-panache-deployment-999-SNAPSHOT-sources.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  14.131 s
[INFO] Finished at: 2024-10-17T11:48:37+02:00
[INFO] ------------------------------------------------------------------------
[INFO] 14 goals, 12 executed, 2 from cache, saving at least 1s
```

**Comment by geoand** [comment]
I assume this is specific to `QuarkusUnitTest`, right?

**Comment by FroMage** [comment]
Probably. Though I can try in dev or test or prod modes as well if you want?

**Comment by geoand** [comment]
Sure yeah, just trying to understand the scope of the problem

**Comment by FroMage** [comment]
Confirmed that in prod/dev/test and continuous testing this does not happen, I do get the proper sql syntax error and startup/test/build error

**Comment by geoand** [comment]
Thanks

**Comment by gsmet** [comment]
This is actually a Flyway issue. They implemented `Exception#getMessage()` and when we try to get the message at the end, the CL is closed and thus the resource can be obtained.

Here is what I obtain after a Flyway fix. I will create a PR but I don't know if/when they will merge it. From what I can see, the community repo doesn't contain the development of the library.

```
[ERROR] io.quarkus.hibernate.orm.panache.deployment.test.FlywayIssue1 -- Time elapsed: 4.630 s <<< ERROR!
java.lang.RuntimeException: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:736)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:305)
	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:689)
	... 1 more
Caused by: org.flywaydb.core.internal.exception.FlywayMigrateException: Script V1.0.0__create.sql failed
--------------------------------
SQL State  : 42000
Error Code : 42000
Message    : Syntax error in SQL statement "create sequence MyEntity_SEQ start with 1 increment by 50 /*semi-colon omitted on purpose*/\000a      [*]create table MyEntity (\000a          id bigint not null,\000a          primary key (id)\000a      )"; SQL statement:
create sequence MyEntity_SEQ start with 1 increment by 50 /*semi-colon omitted on purpose*/
      create table MyEntity (
          id bigint not null,
          primary key (id)
      ) [42000-230]
Location   : database/multitenant/V1.0.0__create.sql (/home/gsmet/git/quarkus/extensions/panache/hibernate-orm-panache/deployment/target/quarkus-unit-test4213525625088269825/app-root/database/multitenant/V1.0.0__create.sql)
Line       : 1
Statement  : Run Flyway with -X option to see the actual statement causing the problem
```

**Comment by gsmet** [comment]
I created https://github.com/flyway/flyway/pull/3968 .

I will close this as I don't think there's anything we can do on the Quarkus side except hoping they will merge this pull request.

**Comment by FroMage** [comment]
Oh, damn. Good investigation! I suppose there's no way for us to obtain the message before we close the CL? I suppose they'll argue that we should not close CLs because we're done processing types that use them?

**Comment by FroMage** [comment]
> I don't think what we're doing is incorrect. Exception will always leak in a way or another if you throw them. Having the getMessage() actually loading resources from the CL is extremely odd and brittle.

It is odd and brittle. But it also feels weird that we're closing the CL that is in use by an exception while it's being thrown/handled. I can see the argument go both ways 🤷

**Comment by gsmet** [comment]
The issue has been fixed in Flyway `11.3.0`
